addon/components/row/table-row.ts

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TableAPI, RowMeta, TableMeta } from '@gavant/ember-table';
+
+interface TableRowArgs {
+    api: TableAPI<any>;
+    rowMeta: RowMeta<any>;
+    tableMeta: TableMeta;
+    onClick?(...args: any[]): void;
+    onDoubleClick?(...args: any[]): void;
+}
+
+enum SELECT_MODE {
+    NONE = 'none',
+    SINGLE = 'single',
+    MULTIPLE = 'multiple'
+}
+
+export default class TableRow extends Component<TableRowArgs> {
+    get rowValue() {
+        return this.args.api.rowValue;
+    }
+
+    get rowMeta() {
+        return this.args.api.rowMeta;
+    }
+
+    get cells() {
+        return this.args.api.cells;
+    }
+
+    get rowSelectionMode() {
+        return this.args.api.rowSelectionMode;
+    }
+
+    get isHeader() {
+        return this.args.api.isHeader;
+    }
+
+    get isSelected() {
+        return this.args.rowMeta.isSelected;
+    }
+
+    get isGroupSelected() {
+        return this.args.rowMeta.isGroupSelected;
+    }
+
+    get isSelectable() {
+        return this.rowSelectionMode === SELECT_MODE.MULTIPLE || this.rowSelectionMode === SELECT_MODE.SINGLE;
+    }
+
+    get isExpanded() {
+        return this.args.tableMeta.expandedRows?.includes(this.rowValue.id) ?? false;
```

Comment:
if we keep this logic as is, i think the `expandedRows` array should contain the entire rowValue objects (so in  most cases, the `DS.Model`s), as some tables row objects may not even have an `id` property, if its just rendering a list of POJOs or something.